### ISSUE TITLE
Add Certificate ID

### ISSUE THREAD

**Issue by JosiahCarberry** [markn86/moodle-mod_customcert#39]
It is common for a training organization to maintain a register of certificates, allowing anyone(?) to validate a particular certificate. This implies that certificates have unique IDs, that the ID is printable on the certificate, and that the ID is related to a student record, which is related to a course record. Naturally, merely reprinting or redownloading a certificate does not change the ID.
Adding this functionality would be nice to have.

**Comment by markn86** [comment]
Well, there is the 'code' element that does this. It will generate a unique code for that user and certificate. Is this what you are talking about or am I misunderstanding?

**Comment by JosiahCarberry** [comment]
When I wrote the above suggestion, I had previously tested the code field and found that nothing displayed at all. But, retesting it following your remark, I see that it displays what appears to be a random 10 character code. I guess that could do the trick, so long as the uniqueness of the code can be guaranteed. 

Since I cannot reproduce the initial problem, I will let that sleeping dog lie.

I wonder if that code value has any check digit in place or other means for validating its authenticity or correctness?

I presume that the xxxcustomcert_issues table is the register of certificates. If so, it seems that the only way to relate the issued certificate to a course is if the certificate has the coursename element in it. Is this correct? I have found that using a text field, rather than coursename, in the certificate gives a needed level of control. The value of coursename used elsewhere in Moodle might not be the value I want to appear on the certificate.
